tcms/report/views.py

```diff
@@ -213,27 +228,27 @@ def get(self, request, product_id):
         except (TypeError, ValueError, Product.DoesNotExist) as error:
             raise Http404(error)
 
-        return super(ProductComponentReport, self).get(request, product_id)
+        return super().get(request, *args, **kwargs)
 
     def get_context_data(self, **kwargs):
         components = self.product.component.select_related('product')
         components = components.only('name', 'product__name')
 
-        pid = self.product.pk
-        total_cases = self.total_cases(pid)
-        failed_case_runs_count = self.failed_case_runs_count(pid)
-        finished_case_runs_count = self.finished_case_runs_count(pid)
+        product_id = self.product.pk
+        total_cases = self.total_cases(product_id)
+        failed_case_runs_count = self.failed_case_runs_count(product_id)
+        finished_case_runs_count = self.finished_case_runs_count(product_id)
 
         for component in components:
-            cid = component.pk
-            component.total_cases = total_cases.get(cid, 0)
-            component.failed_case_run_count = failed_case_runs_count.get(cid,
-                                                                         0)
-
-            n = finished_case_runs_count.get(cid, 0)
-            m = component.total_cases
-            if n and m:
-                component.finished_case_run_percent = round(n * 100.0 / m, 1)
+            component_id = component.pk
+            component.total_cases = total_cases.get(component_id, 0)
+            component.failed_case_run_count = failed_case_runs_count.get(component_id, 0)
+
+            total_runs = finished_case_runs_count.get(component_id, 0)
+            total_cases = component.total_cases
```

Comment:
This is causing a failure in reports (/report/product/1/component/). The problem is that on line 238 you assign a dictionary value to this variable and then on line 244 you execute `.get()` on the dictionary. After the first iteration of the loop the `total_cases` variable is reassigned and it is now of type int, which doesn't have the get method defined. 

To reproduce you need to have at least 1 product with 2 or more components defined and create some test runs for them so you can generate the report.